extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergTableLocationProviderBase.java

```diff
@@ -38,23 +41,27 @@ public IcebergTableLocationProviderBase(
     }
 
     /**
-     * Update the table location provider with the latest snapshot from the catalog.
+     * Update a manually refreshing table location provider with the latest snapshot from the catalog. This will throw
+     * an {@link UnsupportedOperationException} if the table is not manually refreshing.
      */
     public abstract void update();
 
     /**
-     * Update the table location provider with a specific snapshot from the catalog. If the {@code snapshotId} is not
-     * found in the list of snapshots for the table, an {@link IllegalArgumentException} is thrown. The input snapshot
-     * must also be newer (higher in sequence number) than the current snapshot or an {@link IllegalArgumentException}
-     * is thrown.
+     * Update a manually refreshing table location provider with a specific snapshot from the catalog. If the
+     * {@code snapshotId} is not found in the list of snapshots for the table, an {@link IllegalArgumentException} is
```

Comment:
Does Iceberg have a `SnapshotNotFoundException`? If not, should we add one of our own? IAE is OK, though.

Reply:
Iceberg does not produce an error when a snapshot is not matched. Intending to keep these as IAE rather than create an exception used exactly once.